frontend/src/lib/components/learningkit/LearningKit.svelte

```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+	import { Trash2 } from 'lucide-svelte';
+	import ConfirmDialog from '$lib/components/dialogs/ConfirmDialog.svelte';
+	import { _ } from 'svelte-i18n';
+	import { createMutation, useQueryClient } from '@tanstack/svelte-query';
+	import { api } from '$lib/api/apiClient.js';
+	import { deleteLearningKit } from '$lib/api/collections/learningKit.js';
+
+	const client = useQueryClient();
+
+	interface LearningKitProps {
+		title: string;
+		uuid: string;
+		queryKey: string;
+	}
+
+	const { title, uuid, queryKey }: LearningKitProps = $props();
+	let showDeleteDialog = $state(false);
+
+	const deleteKitMutation = createMutation({
+		onSuccess: () => {
+			client.invalidateQueries({ queryKey: [queryKey] });
+		},
+		mutationFn: (kitId: string) => api(fetch).req(deleteLearningKit, null, kitId).parse()
+	});
+
+	async function handleConfirmDelete() {
+		if (!uuid) return;
+
+		await $deleteKitMutation.mutateAsync(uuid);
```

Comment:
For new code I always advice to work with `mutate` and zero `async` code instead. Use the `deleteKitMutation.isLoading` for showing loading states instead.
But wew can keep this, it works